requirements/extra.txt

```diff
@@ -8,3 +8,4 @@ onnx>=1.7.0
 onnxruntime>=1.3.0
 hydra-core>=1.0
 https://github.com/PyTorchLightning/fairscale/archive/pl_1.1.0.zip
+jsonargparse[signatures]>=3.3.1  # LightningCLI requirement
```

Comment:
There were comments saying that we should move this to extras. But since LightningCLI is supposed to be extendable by users it seems important that the api appears in the documentation. If I add a check then when building the docs it wouldn't find the classes so wouldn't be included, no? That is what I understood from the errors in the doc builds, since it seems that to build the docs does not preinstall the extras. Please correct me if I am understanding something wrong.

Reply:
I have modified pytorch_lightning.utilities.cli so that it can be imported even when jsonargparse not available.